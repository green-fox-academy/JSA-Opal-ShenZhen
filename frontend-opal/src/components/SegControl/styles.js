import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  navBars: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  navTitle: {
    padding: 2,
    width: '50%',
    height: 50,
    textAlign: 'center',
    fontSize: 12
  },
  buttonPressed: {
    borderBottomWidth: 1,
    borderBottomColor: '#e17055'
  }
});
