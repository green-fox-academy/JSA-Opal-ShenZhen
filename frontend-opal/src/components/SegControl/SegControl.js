import React from 'react';
import { Button, View } from 'react-native';
import styles from './styles';

const SegControl = () => {
  const navTitles = [
    { id: 1, key: 'PORTFOLIO' },
    { id: 2, key: 'WATCHLIST' }
  ];

  const handlePress = key => key === 'PORTFOLIO';

  const renderItems = titles =>
    titles.map(title => (
      <View
        key={title.id}
        style={[styles.navTitle, handlePress(title.key) && styles.buttonPressed]}
      >
        <Button title={title.key} color="#e17055" />
      </View>
    ));

  return <View style={styles.navBars}>{renderItems(navTitles)}</View>;
};

export default SegControl;
