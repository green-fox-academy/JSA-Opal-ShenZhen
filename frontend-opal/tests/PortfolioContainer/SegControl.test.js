import React from 'react';
import renderer from 'react-test-renderer';
import SegControl from 'components/SegControl/SegControl';

describe('<SegControl />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SegControl />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
