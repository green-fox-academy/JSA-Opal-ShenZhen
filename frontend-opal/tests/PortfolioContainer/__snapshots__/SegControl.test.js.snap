// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SegControl /> renders correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "fontSize": 12,
          "height": 50,
          "padding": 2,
          "textAlign": "center",
          "width": "50%",
        },
        Object {
          "borderBottomColor": "#e17055",
          "borderBottomWidth": 1,
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#e17055",
              },
            ]
          }
        >
          PORTFOLIO
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "fontSize": 12,
          "height": 50,
          "padding": 2,
          "textAlign": "center",
          "width": "50%",
        },
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#e17055",
              },
            ]
          }
        >
          WATCHLIST
        </Text>
      </View>
    </View>
  </View>
</View>
`;
